# ANÁLISIS DE ENCUESTA DE RESTAURANTES DE COMIDA RÁPIDA

Este notebook consolida los 20 analisis realizados sobre el archivo encuesta_restaurantes_10000.csv, en donde se utiliza un diccionario anidado para almacenar la informacion de una encuesta sobre consumo en restaurante.  Cada seccion lee la columna correspondiente y produce su conteo.


In [33]:
import csv #Se importa la libreria poder leer y procesar el archivo de datos

## Diccionario

In [34]:
restaurante = {
    "id": [],
    "preferencias": {"comida": [], "frecuencia": []},
    "consumo": {"gasto": []},
    "experiencia": {
        "producto": [],
        "servicio": [],
        "tiempo": [],
        "precio": [],
    },
    "nps": {
        "recomendacion": [],
        "volveria": [],
        "general": [],
    },
}

separador_de_reportes = "-" * 108

with open("encuesta_restaurantes_10000.csv", "r") as encuestas:
    datos_restaurante = csv.DictReader(encuestas)
    for valor in datos_restaurante:
        restaurante["id"].append(int(valor["id"]))
        restaurante["preferencias"]["comida"].append(valor["comida_preferida"])
        restaurante["preferencias"]["frecuencia"].append(valor["frecuencia_consumo"])
        restaurante["consumo"]["gasto"].append(float(valor["gasto_promedio"]))
        restaurante["experiencia"]["producto"].append(int(valor["satisfaccion_producto"]))
        restaurante["experiencia"]["servicio"].append(int(valor["satisfaccion_servicio"]))
        restaurante["experiencia"]["tiempo"].append(valor["tiempo_entrega"])
        restaurante["experiencia"]["precio"].append(valor["precio_percepcion"])
        restaurante["nps"]["recomendacion"].append(int(valor["recomendaria"]))
        restaurante["nps"]["volveria"].append(valor["volveria_comprar"].strip().lower() == "true")
        restaurante["nps"]["general"].append(int(valor["calificacion_general"]))

---
## Reporte 1: Cantidad de personas por comida preferida 

En este reporte se analiza la cantidad de personas segun su comida rapida preferida con el objetivo de identificar cuales son las opciones mas demandadas por los consumidores


In [35]:
def reporte_1():
    resultado = {}  # Creamos un diccionario vacio donde guardaremos el conteo de cada comida

    for comida in restaurante["preferencias"]["comida"]:    # Recorremos la lista de comidas preferidas de todas las personas
        if comida in resultado:         # comprobamos si esa comida ya existe en el diccionario
            resultado[comida] += 1      # aumentamos 1 si existe 
        else:                           # si la comida aparece por primera vez
            resultado[comida] = 1       # la agregamos al diccionario con valor inicial 1

    return resultado            # devolvemos el diccionario con los conteos

datos = reporte_1()   # guardamos el resultado de la funcion en una variable

print("Cantidad de personas por comida preferida.") 
print("------------------------------------------")

Dados = sorted(datos.items(), key=lambda x: x[1], reverse=True)
# ordenamos el diccionario de mayor a menor segun la cantidad
# datos.items() convierte el diccionario en pares
# lambda x: x[1] nos ordenada por la cantidad de personas 
# reverse=True nos ordenara de mas a menos personas 
for comida, cantidad in Dados:          # recorremos los datos ya ordenados
    print(f"{comida}: {cantidad} personas")    # mostramos cada comida con su cantidad

Cantidad de personas por comida preferida.
------------------------------------------
Pizza: 2053 personas
Hot Dogs: 2032 personas
Hamburguesas: 1997 personas
Pollo Frito: 1992 personas
Tacos: 1926 personas


### Conclusión:
La pizza es la comida mas preferida, seguida de hot dogs y hamburguesas pero se puede ver que no es tanta la diferencia de personas entre una comida y otra 

---
## Reporte 2: Frecuencia de consumo  

En este reporte se analiza con que frecuencia las personas consumen comida rapida con el objetivo de identificar los habitos de consumo de los clientes


In [36]:
def reporte_2():
    resultado = {}  # creamos un diccionario vacio para contar las frecuencias

    for frecuencia in restaurante["preferencias"]["frecuencia"]:  # recorremos la lista de frecuencias
        if frecuencia in resultado:       # verificamos si ya existe en el diccionario
            resultado[frecuencia] += 1    # si existe aumentamos 1
        else:
            resultado[frecuencia] = 1     # si no existe la agregamos con valor 1

    return resultado  # devolvemos el diccionario con los conteos


datos = reporte_2()  # guardamos el resultado en una variable

print("Frecuencia de consumo")
print("---------------------")

ordenado = sorted(datos.items(), key=lambda x: x[1], reverse=True)
# ordenamos de mayor a menor segun la cantidad de personas

for frecuencia, cantidad in ordenado:  # recorremos los datos ordenados
    print(f"{frecuencia}: {cantidad} personas")  # mostramos los resultados

Frecuencia de consumo
---------------------
Semanal: 2534 personas
Quincenal: 2495 personas
Mensual: 2493 personas
Diario: 2478 personas


### Conclusión:
La mayoria de las personas consume comida rapida con una frecuencia regular lo que indica que este tipo de alimento forma parte de sus habitos de consumo

---
## Reporte 3: Promedio de gasto

En este reporte se analiza el gasto promedio de los consumidores con el objetivo de conocer cuanto dinero gastan en comida rapida


In [37]:
def reporte_3():
    datos = restaurante["consumo"]["gasto"]   # obtenemos la lista de gastos de todas las personas
    total = 0

    for gasto in datos:
        total += gasto

    resultado = {
        "gasto": {
            "total": total,     # guardamos el total de todos los gastos
            "cantidad": len(datos),   # guardamos la cantidad de personas
            "promedio": total / len(datos),   # calculamos el promedio dividiendo total entre cantidad
        }
    }

    return resultado

promedio = reporte_3()  # guardamos el resultado

print("Promedio de gasto")
print("-----------------")
print("Promedio: $", round(promedio["gasto"]["promedio"], 2))  # mostramos el promedio con 2 decimales

Promedio de gasto
-----------------
Promedio: $ 69.82


### Conclusion:
El promedio de gasto permite identificar cuanto dinero invierten los consumidores en comida rapida

---
## Reporte 4: Promedio de satisfacción del producto 

En este reporte se analiza el nivel de satisfaccion del producto con el objetivo de conocer la opinion de los clientes


In [38]:
def reporte_4():
    datos = restaurante["experiencia"]["producto"]
    total = 0

    # sumamos las calificaciones del producto
    for valor in datos:
        total += valor

    resultado = {
        "satisfaccion_producto": {
            "promedio": total / len(datos)  # calculamos el promedio
        }
    }

    return resultado  # devolvemos el resultado


promedio = reporte_4()  

print("Promedio de satisfaccion del producto")
print("-------------------------------------")

print("Promedio:", round(promedio["satisfaccion_producto"]["promedio"]), "/ 10")
print("Promedio redondeado a dos decimales:", round(promedio["satisfaccion_producto"]["promedio"], 2), "/ 10")

Promedio de satisfaccion del producto
-------------------------------------
Promedio: 5 / 10
Promedio redondeado a dos decimales: 4.99 / 10


### Conclusion:
Un valor alto indica que los consumidores estan satisfechos mientras que un valor bajo podria reflejar problemas en la calidad o en la experiencia del producto

---
## Reporte 5: Promedio de satisfacción del servicio

En este reporte se analiza el nivel de satisfaccion del servicio con el objetivo de conocer la opinion de los clientes


In [39]:
def reporte_5():
    datos = restaurante["experiencia"]["servicio"]  
    # obtenemos la lista de calificaciones del servicio

    total = 0  
    # variable para acumular la suma

    # sumamos las calificaciones del servicio
    for valor in datos:
        total += valor  

    resultado = {
        "satisfaccion_servicio": {
            "promedio": total / len(datos)  
            # calculamos el promedio
        }
    }

    return resultado  
    # devolvemos el diccionario con el resultado


promedio = reporte_5()  
# guardamos el resultado


print("Promedio de satisfaccion del servicio")
print("-------------------------------------")

print("Promedio:", round(promedio["satisfaccion_servicio"]["promedio"]), "/ 10")
print("Promedio redondeado a dos decimales:", round(promedio["satisfaccion_servicio"]["promedio"], 2), "/ 10")

Promedio de satisfaccion del servicio
-------------------------------------
Promedio: 5 / 10
Promedio redondeado a dos decimales: 4.98 / 10


### Conclusion:
Un valor alto indica que el servicio es bien percibido, mientras que un valor bajo podria señalar problemas en la atencion o en la experiencia del cliente

---
## Reporte 6: Distribución del tiempo de entrega


Cuenta cuantos pedidos fueron clasificados como Rapido, Aceptable o Lento, y calcula el porcentaje de cada uno sobre el total.

In [40]:
def reporte_6():
    print(f"Reporte 6: distribucion del tiempo de entrega de pedidos")
    print(separador_de_reportes)
    tiempos_entrega = {"Rapido": 0, "Aceptable": 0, "Lento": 0}
    for conteo in restaurante["experiencia"]["tiempo"]:
        if conteo in tiempos_entrega:
            tiempos_entrega[conteo] += 1
    pedidos_rapido = tiempos_entrega["Rapido"]
    pedidos_aceptable = tiempos_entrega["Aceptable"]
    pedidos_lento = tiempos_entrega["Lento"]
    total_pedidos = pedidos_aceptable + pedidos_rapido + pedidos_lento
    porcentaje_pedidos_rapidos = (pedidos_rapido / total_pedidos) * 100
    porcentaje_pedidos_aceptables = (pedidos_aceptable / total_pedidos) * 100
    porcentaje_pedidos_lentos = (pedidos_lento / total_pedidos) * 100
    print(f"""
    total pedidos rapidos: {pedidos_rapido},
    total pedidos aceptables: {pedidos_aceptable},
    total pedidos lentos: {pedidos_lento}
    """)
    print(f"""
    porcentaje pedidos rapidos: {porcentaje_pedidos_rapidos}%,
    porcentaje pedidos aceptables: {porcentaje_pedidos_aceptables}%,
    porcentaje pedidos lentos: {porcentaje_pedidos_lentos}%
    """)
    print(separador_de_reportes)

reporte_6()

Reporte 6: distribucion del tiempo de entrega de pedidos
------------------------------------------------------------------------------------------------------------

    total pedidos rapidos: 0,
    total pedidos aceptables: 3356,
    total pedidos lentos: 3337
    

    porcentaje pedidos rapidos: 0.0%,
    porcentaje pedidos aceptables: 50.14193933960854%,
    porcentaje pedidos lentos: 49.85806066039145%
    
------------------------------------------------------------------------------------------------------------


---
## Reporte 7: Distribución de percepción de precios 


Cuenta cuantos clientes perciben el precio como Alto, Medio o Bajo, y calcula el porcentaje de cada categoria.

In [41]:
def reporte_7():
    print("Reporte 7: distribucion de percepcion de precios")
    print(separador_de_reportes)
    percepcion_precios = {"Alto": 0, "Medio": 0, "Bajo": 0}
    experiencia_precio = restaurante["experiencia"]["precio"]
    for conteo in experiencia_precio:
        if conteo in percepcion_precios:
            percepcion_precios[conteo] += 1
    percepcion_precios_alto = percepcion_precios["Alto"]
    percepcion_precios_medio = percepcion_precios["Medio"]
    percepcion_precios_bajo = percepcion_precios["Bajo"]
    total_percepciones = percepcion_precios_alto + percepcion_precios_medio + percepcion_precios_bajo
    porcentaje_precio_alto = (percepcion_precios_alto / total_percepciones) * 100
    porcentaje_precio_medio = (percepcion_precios_medio / total_percepciones) * 100
    porcentaje_precio_bajo = (percepcion_precios_bajo / total_percepciones) * 100
    print(f"""
    personas que perciben precios altos: {percepcion_precios_alto}, 
    personas que perciben precios medios: {percepcion_precios_medio},
    personas que perciben precios bajos: {percepcion_precios_bajo}
    """)
    print(f"""
    porcentaje de personas que perciben precios altos: {porcentaje_precio_alto}%,
    porcentaje de personas que perciben precios medios: {porcentaje_precio_medio}%,
    porcentaje de personas que perciben precios bajos: {porcentaje_precio_bajo}%
    """)
    print(separador_de_reportes)

reporte_7()

Reporte 7: distribucion de percepcion de precios
------------------------------------------------------------------------------------------------------------

    personas que perciben precios altos: 3386, 
    personas que perciben precios medios: 3243,
    personas que perciben precios bajos: 3371
    

    porcentaje de personas que perciben precios altos: 33.86%,
    porcentaje de personas que perciben precios medios: 32.43%,
    porcentaje de personas que perciben precios bajos: 33.71%
    
------------------------------------------------------------------------------------------------------------


---
## Reporte 8: Promedio general de satisfacción


Calcula el promedio de la columna `recomendaria` como indicador de satisfaccion general.

In [42]:
def reporte_8():
    print("Reporte 8: promedio de satisfaccion general")
    print(separador_de_reportes)
    recomendacion = restaurante["nps"]["recomendacion"]
    print(f"""
    promedio de satisfaccion general: {sum(recomendacion) / len(recomendacion):.2f}
    """)
    print(separador_de_reportes)

reporte_8()

Reporte 8: promedio de satisfaccion general
------------------------------------------------------------------------------------------------------------

    promedio de satisfaccion general: 5.00
    
------------------------------------------------------------------------------------------------------------


---
## Reporte 9: Porcentaje de clientes que volverían 


Cuenta cuantos clientes respondieron True o False en `volveria_comprar` y calcula el porcentaje de cada grupo.

In [43]:
def reporte_9():
    print("Reporte 9: porcentaje de clientes que volverian")
    print(separador_de_reportes)
    clientes_volverian = restaurante["nps"]["volveria"]
    volverian = {"si": 0, "no": 0}
    for valor in clientes_volverian:
        if valor == True:
            volverian["si"] += 1
        if valor == False:
            volverian["no"] += 1
    clientes_que_volverian = volverian["si"]
    clientes_que_no_volverian = volverian["no"]
    total = clientes_que_no_volverian + clientes_que_volverian
    porcentaje_volverian = (clientes_que_volverian / total) * 100
    porcentaje_no_volverian = (clientes_que_no_volverian / total) * 100
    print(f""" 
    cantidad de clientes que volverian: {clientes_que_volverian}
    cantidad de clientes que no volverian: {clientes_que_no_volverian}
    porcentaje de clientes que regresarian: {porcentaje_volverian:.2f}%
    porcentaje de clientes que no regresarian: {porcentaje_no_volverian:.2f}
    """)
    print(separador_de_reportes)

reporte_9()

Reporte 9: porcentaje de clientes que volverian
------------------------------------------------------------------------------------------------------------
 
    cantidad de clientes que volverian: 4996
    cantidad de clientes que no volverian: 5004
    porcentaje de clientes que regresarian: 49.96%
    porcentaje de clientes que no regresarian: 50.04
    
------------------------------------------------------------------------------------------------------------


---
## Reporte 10: Cálculo del NPS


Clasifica a los encuestados segun su puntuacion en `recomendaria`:

| Grupo | Rango |
|---|---|
| Promotores | 9 - 10 |
| Pasivos | 7 - 8 |
| Detractores | 0 - 6 |

Formula: `NPS = %Promotores - %Detractores`

El resultado va de -100 a +100. Un valor positivo indica mas promotores que detractores.

In [44]:
def reporte_10():
    print("Reporte 10: NPS (Net Promoter Score)")
    print(separador_de_reportes)
    recomendacion = restaurante["nps"]["recomendacion"]
    total = len(recomendacion)
    conteo = {"promotores": 0, "pasivos": 0, "detractores": 0}
    for recomendados in recomendacion:
        if recomendados >= 9:
            conteo["promotores"] += 1
        elif recomendados >= 7:
            conteo["pasivos"] += 1
        else:
            conteo["detractores"] += 1
    porcentajes = {
        "promotores": (conteo["promotores"] / total) * 100,
        "pasivos": (conteo["pasivos"] / total) * 100,
        "detractores": (conteo["detractores"] / total) * 100,
    }
    nps = porcentajes["promotores"] - porcentajes["detractores"]
    print(f"""
    promedio de NPS: {nps}
    """)
    print(separador_de_reportes)

reporte_10()

Reporte 10: NPS (Net Promoter Score)
------------------------------------------------------------------------------------------------------------

    promedio de NPS: -45.81
    
------------------------------------------------------------------------------------------------------------


---
## Reporte 11: Segmentación: promotores, pasivos, detractores 


En este reporte se clasifica a los clientes en promotores, pasivos y detractores según su calificación, con el objetivo de medir la satisfaccion del cliente.

In [45]:
def reporte_11():
    promotores = 0
    pasivos = 0 
    detractores = 0

    for i in restaurante["nps"]["recomendacion"]:
        if i >= 9:
            promotores += 1
        elif i >= 7:
            pasivos += 1
        else:
            detractores += 1

    total = promotores + pasivos + detractores
    porcentaje_promotores = round((promotores / total) * 100, 2)
    porcentaje_pasivos = round((pasivos / total) * 100, 2)
    porcentaje_detractores = round((detractores / total) * 100, 2)
    
    print(f"Promotores: {promotores} ({porcentaje_promotores}%)")
    print(f"Pasivos: {pasivos} ({porcentaje_pasivos}%)")
    print(f"Detractores: {detractores} ({porcentaje_detractores}%)")

reporte_11()

Promotores: 1811 (18.11%)
Pasivos: 1797 (17.97%)
Detractores: 6392 (63.92%)


---
## Reporte 12: Comida con mayor satisfacción


En este reporte el objetivo principal es idetificar cual es el tipo de comida que tiene una mejor aceptacion por parte de los clientes.

In [46]:
def reporte_12():
    suma = {}
    conteo = {}

    for i in range(len(restaurante["preferencias"]["comida"])):
        comida = restaurante["preferencias"]["comida"][i]
        satisfaccion = restaurante["experiencia"]["producto"][i]

        if comida in suma:
            suma[comida] += satisfaccion
            conteo[comida] += 1
        else:
            suma[comida] = satisfaccion
            conteo[comida] = 1

    mayor_promedio = 0
    mejor_comida = ""

    for comida in suma:
        promedio = suma[comida] / conteo[comida]
        if promedio > mayor_promedio:
            mayor_promedio = promedio
            mejor_comida = comida

    print(f"La comida con mejor experiencia es: {mejor_comida} ({mayor_promedio:.2f})")

reporte_12()

La comida con mejor experiencia es: Hot Dogs (5.08)


---
## Reporte 13: Comida con menor satisfacción 


En este reporte el objetivo principal es idetificar cual es el tipo de comida que tiene la peor aceptacion por parte de los clientes.

In [47]:
def reporte_13():
    suma = {}
    conteo = {}

    for i in range(len(restaurante["preferencias"]["comida"])):
        comida = restaurante["preferencias"]["comida"][i]
        satisfaccion = restaurante["experiencia"]["producto"][i]

        if comida in suma:
            suma[comida] += satisfaccion
            conteo[comida] += 1
        else:
            suma[comida] = satisfaccion
            conteo[comida] = 1

    menor_promedio = 100
    peor_comida = ""

    for comida in suma:
        promedio = suma[comida] / conteo[comida]
        if promedio < menor_promedio:
            menor_promedio = promedio
            peor_comida = comida

    print(f"La comida con peor experiencia es: {peor_comida} ({menor_promedio:.2f})")

reporte_13()

La comida con peor experiencia es: Pizza (4.86)


---
## Reporte 14: Relación entre gasto y satisfacción


Con este reporte se busca identificar la satisfaccion que tienen los clientes con los productos comparado con lo que gasten en ellos 

In [48]:
def reporte_14():
    suma = {"bajo": 0, "medio": 0, "alto": 0}
    conteo = {"bajo": 0, "medio": 0, "alto": 0}

    for i in range(len(restaurante["consumo"]["gasto"])):
        gasto = restaurante["consumo"]["gasto"][i]
        satisfaccion = restaurante["experiencia"]["producto"][i]

        if gasto <= 20:
            suma["bajo"] += satisfaccion
            conteo["bajo"] += 1
        elif gasto <= 50:
            suma["medio"] += satisfaccion
            conteo["medio"] += 1
        else:
            suma["alto"] += satisfaccion
            conteo["alto"] += 1

    print("Promedio de satisfacción por nivel de gasto")

    for categoria in suma:
        if conteo[categoria] > 0:
            promedio = suma[categoria] / conteo[categoria]
            print(f"{categoria}: {promedio:.2f}")

reporte_14()

Promedio de satisfacción por nivel de gasto
bajo: 4.75
medio: 4.95
alto: 5.00


---
## Reporte 15: Frecuencia vs satisfacción 


Con este reporte se busca identificar si la freceuncia con la que los clientes visitan el establecimiento tiene relacion con el nivel de satisfacción en cada visita.

In [49]:
def reporte_15():
    suma = {}
    conteo = {}

    for i in range(len(restaurante["preferencias"]["frecuencia"])):
        frecuencia = restaurante["preferencias"]["frecuencia"][i]
        satisfaccion = restaurante["experiencia"]["producto"][i]

        if frecuencia in suma:
            suma[frecuencia] += satisfaccion
            conteo[frecuencia] += 1
        else:
            suma[frecuencia] = satisfaccion
            conteo[frecuencia] = 1

    print("Promedio de satisfacción por frecuencia")

    for categoria in sorted(suma):
        if conteo[categoria] > 0:
            promedio = suma[categoria] / conteo[categoria]
            print(f"{categoria}: {promedio:.2f}")

reporte_15()

Promedio de satisfacción por frecuencia
Diario: 5.01
Mensual: 4.97
Quincenal: 4.95
Semanal: 5.01


---
## Reporte 16: Precio vs recomendación 


En este reporte se analiza cómo influye la percepción del precio (bajo, medio, alto) en la probabilidad de que un cliente recomiende el restaurante, con el objetivo de identificar si el costo es un factor determinante en la lealtad del consumidor.

In [50]:
# Precio vs recomendación  
def reporte_16():
    print("\nReporte 16: Precio vs Recomendación")
    precios = restaurante["experiencia"]["precio"]
    recomendaciones = restaurante["nps"]["recomendacion"]

    # Agrupar las sumas y conteo de precios
    resumen = {}
    for precio, rec in zip(precios, recomendaciones):
        if precio not in resumen:
            resumen[precio] = {"suma": 0, "cantidad": 0}
        resumen[precio]["suma"] += rec
        resumen[precio]["cantidad"] += 1

    # Imprimir promedios
    for precio, datos in resumen.items():
        promedio = datos["suma"] / datos["cantidad"]
        print(f"Precio: {precio.capitalize():<10} | Promedio de Recomendación: {promedio:.2f} / 10")

reporte_16()


Reporte 16: Precio vs Recomendación
Precio: Bajo       | Promedio de Recomendación: 4.98 / 10
Precio: Alto       | Promedio de Recomendación: 5.01 / 10
Precio: Medio      | Promedio de Recomendación: 5.01 / 10


La percepción del precio no tiene un impacto significativo en el nivel de recomendación del cliente. Ya sea que el cliente perciba el precio como Bajo (4.98), Medio (5.01) o Alto (5.01), el promedio de recomendación se mantiene constante en la mitad exacta de la escala (5/10). Esto sugiere que el costo no es el factor que está frenando o impulsando la lealtad y el marketing de boca en boca.

---
## Reporte 17:  Tiempo de entrega vs satisfacción 


Este reporte estudia la relación entre la rapidez del servicio y la calificación general otorgada por los usuarios para evaluar qué tan crítico es el tiempo de espera en la experiencia final del cliente.

In [51]:
# Tiempo de entrega vs satisfacción  
def reporte_17():
    print("\nReporte 17: Tiempo de Entrega vs Satisfacción General")
    tiempos = restaurante["experiencia"]["tiempo"]
    satisfaccion = restaurante["nps"]["general"]

    resumen = {}
    for tiempo, sat in zip(tiempos, satisfaccion):
        if tiempo not in resumen:
            resumen[tiempo] = {"suma": 0, "cantidad": 0}
        resumen[tiempo]["suma"] += sat
        resumen[tiempo]["cantidad"] += 1

    for tiempo, datos in resumen.items():
        promedio = datos["suma"] / datos["cantidad"]
        print(f"Tiempo de entrega: {tiempo.capitalize():<10} | Satisfacción General Promedio: {promedio:.2f} / 10")
        
reporte_17()


Reporte 17: Tiempo de Entrega vs Satisfacción General
Tiempo de entrega: Aceptable  | Satisfacción General Promedio: 5.01 / 10
Tiempo de entrega: Lento      | Satisfacción General Promedio: 5.02 / 10
Tiempo de entrega: Rápido     | Satisfacción General Promedio: 4.97 / 10


El tiempo de entrega no muestra una correlación lógica con la satisfacción. Los pedidos con entrega calificada como Lento obtuvieron una satisfacción superior (5.02) a los calificados como Rápido (4.97). Al mantenerse todos los valores alrededor de 5.0, se concluye que la rapidez del servicio no está siendo un diferenciador clave para la experiencia del cliente.

---
## Reporte 18: Ranking de comidas más consumidas


En este reporte se presenta un listado ordenado de los platos preferidos por los clientes, permitiendo identificar los productos estrella del menú y las tendencias de consumo actuales.

In [52]:
def reporte_18():
    print("\nReporte 18: Ranking de Comidas Más Consumidas")
    comidas = restaurante["preferencias"]["comida"]

    # Contar cuántas veces se repite cada comida
    conteo = {}
    for comida in comidas:
        conteo[comida] = conteo.get(comida, 0) + 1

    # Ordenar el diccionario de mayor a menor según su cantidad
    ranking = sorted(conteo.items(), key=lambda x: x[1], reverse=True)

    for i, (comida, cantidad) in enumerate(ranking, 1):
        print(f"{i}. {comida:<15} | Consumida por: {cantidad} clientes")

reporte_18()


Reporte 18: Ranking de Comidas Más Consumidas
1. Pizza           | Consumida por: 2053 clientes
2. Hot Dogs        | Consumida por: 2032 clientes
3. Hamburguesas    | Consumida por: 1997 clientes
4. Pollo Frito     | Consumida por: 1992 clientes
5. Tacos           | Consumida por: 1926 clientes


La Pizza es el producto estrella del restaurante, seguida muy de cerca por los Hot Dogs y las Hamburguesas. Sin embargo, el hallazgo más importante es la distribución altamente equilibrada de las ventas, esto indica que el restaurante tiene un menú muy balanceado.

---
## Reporte 19: Promedio general por tipo de comida 


Aquí se desglosa el nivel de satisfacción general según la categoría de alimento consumido, con el fin de determinar qué platillos cumplen mejor con las expectativas de calidad y cuáles requieren mejoras.

In [53]:
# Promedio general por tipo de comida 
def reporte_19():
    print("\nReporte 19: Promedio General por Tipo de Comida")
    comidas = restaurante["preferencias"]["comida"]
    satisfaccion = restaurante["nps"]["general"]

    resumen = {}
    for comida, sat in zip(comidas, satisfaccion):
        if comida not in resumen:
            resumen[comida] = {"suma": 0, "cantidad": 0}
        resumen[comida]["suma"] += sat
        resumen[comida]["cantidad"] += 1

    promedios = []
    for comida, datos in resumen.items():
        promedio = datos["suma"] / datos["cantidad"]
        promedios.append((comida, promedio))

    # Ordenar por promedio más alto
    promedios.sort(key=lambda x: x[1], reverse=True)

    for comida, prom in promedios:
        print(f"Comida: {comida:<15} | Satisfacción General Promedio: {prom:.2f} / 10")

reporte_19()


Reporte 19: Promedio General por Tipo de Comida
Comida: Pizza           | Satisfacción General Promedio: 5.05 / 10
Comida: Hamburguesas    | Satisfacción General Promedio: 5.03 / 10
Comida: Tacos           | Satisfacción General Promedio: 4.99 / 10
Comida: Hot Dogs        | Satisfacción General Promedio: 4.96 / 10
Comida: Pollo Frito     | Satisfacción General Promedio: 4.96 / 10


La satisfacción general no varía mucho dependiendo del platillo elegido. La Pizza y las Hamburguesas lideran ligeramente la satisfacción con 5.05 y 5.03, mientras que los Hot Dogs y el Pollo Frito presentan las calificaciones un poco más bajas (4.96). Pero, al estar todos los promedios parados en el punto medio de la escala, se evidencia que la calidad de la comida es consistente, pero no extraordinaria. 

---
## Reporte 20: Perfil del cliente promedio


Este reporte sintetiza las variables más frecuentes (modas) y los valores medios (promedios) de toda la encuesta para construir un perfil representativo del cliente típico, integrando sus hábitos, gastos y nivel de agrado.

In [54]:
# Perfil del cliente promedio
def reporte_20():
    print("\n--- Reporte 20: Perfil del Cliente Promedio ---")
    total_clientes = len(restaurante["id"])
    
    if total_clientes == 0:
        print("No hay datos cargados para analizar.")
        return

    # Funciones internas para no repetir código
    def calcular_promedio(lista):
        return sum(lista) / len(lista)

    def obtener_moda(lista):
        conteo = {}
        for item in lista:
            conteo[item] = conteo.get(item, 0) + 1
        # Obtener la llave con el valor más alto
        return max(conteo.items(), key=lambda x: x[1])[0]

    # Cálculos estadísticos
    comida_favorita = obtener_moda(restaurante["preferencias"]["comida"])
    frecuencia_tip = obtener_moda(restaurante["preferencias"]["frecuencia"])
    gasto_prom = calcular_promedio(restaurante["consumo"]["gasto"])
    sat_producto_prom = calcular_promedio(restaurante["experiencia"]["producto"])
    sat_servicio_prom = calcular_promedio(restaurante["experiencia"]["servicio"])
    tiempo_tipico = obtener_moda(restaurante["experiencia"]["tiempo"])
    precio_tipico = obtener_moda(restaurante["experiencia"]["precio"])
    recomienda_prom = calcular_promedio(restaurante["nps"]["recomendacion"])
    porcentaje_retorno = (sum(restaurante["nps"]["volveria"]) / total_clientes) * 100

    # Imprimir el perfil final
    print(f"Comida Preferida             : {comida_favorita}")
    print(f"Frecuencia de Consumo Típica : {frecuencia_tip}")
    print(f"Gasto Promedio               : ${gasto_prom:.2f}")
    print(f"Tiempo de Entrega Común      : {tiempo_tipico}")
    print(f"Percepción de Precio Típica  : {precio_tipico}")
    print(f"Promedio Satisfacción Prod.  : {sat_producto_prom:.2f} / 10")
    print(f"Promedio Satisfacción Serv.  : {sat_servicio_prom:.2f} / 10")
    print(f"Promedio Recomendación (NPS) : {recomienda_prom:.2f} / 10")
    print(f"Probabilidad de Retorno      : {porcentaje_retorno:.1f}% de clientes")
reporte_20()


--- Reporte 20: Perfil del Cliente Promedio ---
Comida Preferida             : Pizza
Frecuencia de Consumo Típica : Semanal
Gasto Promedio               : $69.82
Tiempo de Entrega Común      : Aceptable
Percepción de Precio Típica  : Alto
Promedio Satisfacción Prod.  : 4.99 / 10
Promedio Satisfacción Serv.  : 4.98 / 10
Promedio Recomendación (NPS) : 5.00 / 10
Probabilidad de Retorno      : 50.0% de clientes


El cliente promedio del restaurante tiene un consumo semanal, gasta alrededor de $69.82 y tiene preferencia por la Pizza. Este usuario típico percibe los precios como Altos y los tiempos de entrega como Aceptables. Eso sí, no lo recomendaría a otros. En consecuencia, el restaurante se enfrenta a una fidelización dividida, con solo un 50.0% de probabilidad de retorno.